moveit_core/collision_detection_fcl/src/collision_common.cpp

```diff
@@ -511,17 +518,14 @@ bool distanceCallback(fcl::CollisionObjectd* o1, fcl::CollisionObjectd* o2, void
         return cdata->done;
       }
     }
-    else if (cdata->req->type == DistanceRequestType::GLOBAL)
-    {
-      dist_threshold = cdata->res->minimum_distance.distance;
-    }
     else if (cdata->req->type == DistanceRequestType::SINGLE)
     {
       dist_threshold = it->second[0].distance;
     }
   }
 
-  fcl::DistanceResultd fcl_result;
+  thread_local fcl::DistanceResultd fcl_result;
```

Comment:
This struct is completely allocated on stack as well. No need for `thread_local` storage.